spec/python_webapp_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'serverspec'
+require 'docker'
+
+image = Docker::Image.build_from_dir('python_webapp/')
+
+set :os, family: :redhat
+set :backend, :docker
+set :docker_image, image.id
+
+describe 'Docker configuration' do
+  it 'should expose port 80' do
+    expect(image.json['Config']['ExposedPorts']).to include('80/tcp')
+  end
+end
+
+%w(gcc python-devel python-setuptools libffi-devel nginx).each do |pkg|
+  describe package(pkg) do
+    it { should be_installed }
+  end
+end
+
+describe port(80) do
+  it { should be_listening.with('tcp') }
+end
+
+describe command('pip') do
+  its(:exit_status) { should eq 0 }
+end
+
+%w(gunicorn supervisor).each do |pkg|
+  describe package(pkg) do
+    it { should be_installed.by('pip') }
+  end
+end
+
+describe command('cat /etc/redhat-release') do
+  its(:stdout) { should contain 'CentOS release 6' }
```

Comment:
![Test All the Things!](http://www.daedtech.com/wp-content/uploads/2012/12/TestAllTheThings.jpg)
but seriously, this seems a little excessive.